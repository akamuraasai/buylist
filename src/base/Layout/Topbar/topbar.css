.Topbar {
    position: fixed;
    top: 0;
    height: 60px;
    width: 100%;
    background-color: #fefefe;
    display: inline-flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    box-shadow: 0px 1px 1px 0px #cbcbcb;
    text-align: center;
}

.Topbar__item {
    width: 15%;
}

.Topbar__title {
    width: 70%;
}