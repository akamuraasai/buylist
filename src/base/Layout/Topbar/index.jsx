import React from 'react';
import './topbar.css';

const Template = () => (
  <div className="Topbar">
    <div className="Topbar__item">
      <i className="fa fa-bars"></i>
    </div>
    <div className="Topbar__title">
      <h1>Home</h1>
    </div>
    <div className="Topbar__item">
      <i className="fa fa-plus"></i>
    </div>
  </div>
);

export default Template;
