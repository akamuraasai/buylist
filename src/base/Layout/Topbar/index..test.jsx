import React from 'react';
import { shallow } from 'enzyme';
import Topbar from './index';

describe('Topbar', () => {
  const topbar = shallow(<Topbar />);

  it('should render without crashing', () => {
    expect(topbar).toMatchSnapshot();
  });
});
