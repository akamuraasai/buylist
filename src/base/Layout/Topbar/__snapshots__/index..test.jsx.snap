// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Template />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Topbar__item"
>
          <i
                    className="fa fa-bars"
          />
</div>,
        <div
          className="Topbar__title"
>
          <h1>
                    Home
          </h1>
</div>,
        <div
          className="Topbar__item"
>
          <i
                    className="fa fa-plus"
          />
</div>,
      ],
      "className": "Topbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa fa-bars"
/>,
          "className": "Topbar__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-bars",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Home
</h1>,
          "className": "Topbar__title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home",
          },
          "ref": null,
          "rendered": "Home",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa fa-plus"
/>,
          "className": "Topbar__item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-plus",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Topbar__item"
>
            <i
                        className="fa fa-bars"
            />
</div>,
          <div
            className="Topbar__title"
>
            <h1>
                        Home
            </h1>
</div>,
          <div
            className="Topbar__item"
>
            <i
                        className="fa fa-plus"
            />
</div>,
        ],
        "className": "Topbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-bars"
/>,
            "className": "Topbar__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-bars",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Home
</h1>,
            "className": "Topbar__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Home",
            },
            "ref": null,
            "rendered": "Home",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-plus"
/>,
            "className": "Topbar__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-plus",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
